### ISSUE TITLE
applying 2.4.0 with the plugins DSL fails

### ISSUE THREAD

**Issue by TheBoegl** [TheBoegl/gradle-launch4j#60]
Applying the plugin in version `2.4.0` with the plugins DSL
```gradle
plugins {
   id 'edu.sc.seis.launch4j' version '2.4.0'
}
```
does not work.
This fails with the following error: 
```
Error:Could not find com.springsource.org.apache.batik.ext.awt.jar (org.apache.batik:com.springsource.org.apache.batik.ext.awt:1.7.0).
Searched in the following locations:
    https://jcenter.bintray.com/org/apache/batik/com.springsource.org.apache.batik.ext.awt/1.7.0/com.springsource.org.apache.batik.ext.awt-1.7.0.jar
```

**Comment by TheBoegl** [comment]
Applying the plugin does work...
```gradle

buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'edu.sc.seis.gradle:launch4j:2.4.0'
    }
}
apply plugin: 'edu.sc.seis.launch4j'
```

**Comment by javydreamercsw** [comment]
After your proposed change I get this error:

> FAILURE: Build failed with an exception.
> 
> * Where:
> Build file 'C:\Users\ortizj\Documents\NetBeansProjects\pcgen\build.gradle' line: 222
> 
> * What went wrong:
> A problem occurred evaluating root project 'pcgen'.
> > Could not get unknown property 'cleanCreateExe' for root project 'pcgen' of type org.gradle.api.Project.

**Comment by TheBoegl** [comment]
I found an interesting behavior: After deleting ~/.gradle/caches/modules-2/metadata-2.23 applying this plugin with the plugins DSL works again.

After that either way to apply this plugin works on your __Release_Bug_Fixes__ branch.
Can you confirm that?

**Comment by javydreamercsw** [comment]
I'm away of my programing machine until tomorrow. 

Do you mean just changing the version to 2.4.0, without any of your work arounds, works after cleaning the cache?

**Comment by henriIV** [comment]
I don't know if this could help. 
I was able to resolve the 404 issue in the following way 
Instead of using the suggested buildscript on https://plugins.gradle.org/plugin/edu.sc.seis.launch4j  
buildscript {
  repositories {
    maven {
      url "https://plugins.gradle.org/m2/"
    }
  }
  dependencies {
    classpath "gradle.plugin.edu.sc.seis.gradle:launch4j:2.4.0"
  }
}

**Comment by javydreamercsw** [comment]
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'pcgen'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find com.springsource.org.apache.batik.ext.awt.jar (org.apache.batik:com.springsource.org.apache.batik.ext.awt:1.7.0).
     Searched in the following locations:
         https://plugins.gradle.org/m2/org/apache/batik/com.springsource.org.apache.batik.ext.awt/1.7.0/com.springsource.org.apache.batik.ext.awt-1.7.0.jar
   > Could not find com.springsource.org.apache.batik.util.jar (org.apache.batik:com.springsource.org.apache.batik.util:1.7.0).
     Searched in the following locations:
         https://plugins.gradle.org/m2/org/apache/batik/com.springsource.org.apache.batik.util/1.7.0/com.springsource.org.apache.batik.util-1.7.0.jar

If you follow those links the files are just not there.

**Comment by TheBoegl** [comment]
These files/dependencies are on the exclusion list in launch4j's pom and should not be retrieved at all.
This seems to me like gradle's cache is not working as expected.

**Comment by breuniga** [comment]
I did get the same error message, but did not use the plugin DSL. Cleaning the cache did not help. Then I recognised that I forgot to change
```
buildscript { ...
   dependencies {
      classpath 'gradle.plugins.edu.sc.seis.gradle:launch4j:2.4.0'
   }
}
```
to
```
buildscript { ...
   dependencies {
      classpath 'edu.sc.seis.gradle:launch4j:2.4.0'
   }
}
```
After correcting this, everything worked fine.

**Comment by javydreamercsw** [comment]
That's not the issue on my case. I still believe there's something wrong.

I use gradle 4. maybe that's the difference?

**Comment by TheBoegl** [comment]
@javydreamercsw In fa632f8014 I used some launch4j classes directly instead of starting a new jvm and calling `launch4jc`. Therefore the version _2.4.0_ has a compile dependency to `launch4j`. Its [pom
](http://central.maven.org/maven2/net/sf/launch4j/launch4j/3.11/launch4j-3.11.pom) clearly exclude the erroneous `com.springsource.org.apache.batik.ext.awt` artifact.

I'm able to reproduce this issue if I revert the _metadata-23_ cache files from a backup.
Refreshing the cache with `./gradlew createExe --refresh-dependencies` resolves this issue with gradle versions _3.4_, _3.5_, _4.0_ and _4.0.2_ on two test machines here.

**Comment by javydreamercsw** [comment]
I'll give it a try tomorrow and report Back. I'll have to add notes about it I believe.

**Comment by javydreamercsw** [comment]
Yup, it worked fine now. maybe it's worth adding to the documentation in readme or something.

**Comment by TheBoegl** [comment]
This issue was produced by the plugin-publish plugin which does not handle the excludes correctly. 
The work around is to set the module to `module: '*'` instead of leaving it out.

**Comment by javydreamercsw** [comment]
So will this be a new version release, or will just re-publish under same version?

**Comment by TheBoegl** [comment]
The important commit is published under version [2.4.1](https://github.com/TheBoegl/gradle-launch4j/releases/tag/v2.4.1).

**Comment by javydreamercsw** [comment]
Confirmed, it works like a charm! Thanks!
